### PR TITLE
[gradle] Add Spec

### PR DESCRIPTION
**Pull request by Monkeyank**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature
**What is the new behavior (if this is a feature change)?**
Adds `gradle` spec. (Although `./gradlew` doesn't work 😢)

### CODE REVIEW

File: dev/gradle.ts [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+    },
+    {
+      name: "rich",
+      description:
+        "Enables color and other rich output in the console output, regardless of whether the build process is not attached to a console",
+    },
+    {
+      name: "verbose",
+      description:
+        "enable color and other rich output like the rich, but output task names and outcomes at the lifecycle log level, as is done by default in Gradle 3.5 and earlier",
```
Comment (QuiiBz) [inline dev/gradle.ts:23]:
```suggestion
        "Enable color and other rich output like the rich, but output task names and outcomes at the lifecycle log level, as is done by default in Gradle 3.5 and earlier",
```

File: dev/gradle.ts [hunk 2]
```diff
@@ -0,0 +1,538 @@
...
+    args: {
+      name: "directory",
+      template: "folders",
+    },
+  },
+];
+
+export const completion: Fig.Spec = {
+  // ./gradlew doesn't work
+  // name: ["gradle", "./gradlew"],
```
Comment (QuiiBz) [inline dev/gradle.ts:354]: I think gradlew should be in a separate spec

File: dev/gradle.ts [hunk 3]
```diff
@@ -0,0 +1,538 @@
...
+          description:
+            "Gradle will execute every task to be executed where all of the dependencies for that task completed without failure",
+        },
+        ...sharedOptions,
+      ],
+    },
+  ],
+  options: [...sharedOptions],
+
+  // There isn't a generator for gradle tasks (takes too long)
```
Comment (QuiiBz) [inline dev/gradle.ts:532]: We can read the `build.gradle` file and put a regex to find all the tasks - I don't think that would be slow

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Entering the Java world! Almost a perfect PR, but I think we should add a generator for the tasks. See the comment on this below

File: dev/gradle.ts (hunk 3 above)
Comment (Monkeyank) [reply dev/gradle.ts:532]: I don't think the build.gradle contains all the tasks. At least based on some of my projects that I've checked out

Comment (QuiiBz) [reply dev/gradle.ts:530]: Yes, we can add the default tasks as suggestions and additional tasks from the build.gradle file using a generator

**Review by QuiiBz — APPROVED** [review]
LGTM!

**Pull request merged**